Hotrod/character.py

```diff
@@ -0,0 +1,655 @@
+
+                elif self.current_direction == direction.Direction.up:
+                    if ((self.center_y >= current_cell.center_y and current_cell.center_y > previous_y) or
+                            (self.center_y == current_cell.center_y and previous_y == current_cell.center_y)):
+                        self.center_y = current_cell.center_y
+                        self._set_direction()
+
+                elif self.current_direction == direction.Direction.down:
+                    if ((self.center_y <= current_cell.center_y and current_cell.center_y < previous_y) or
+                            (self.center_y == current_cell.center_y and previous_y == current_cell.center_y)):
+                        self.center_y = current_cell.center_y
+                        self._set_direction()
+
+    def _set_direction(self):
+        """Set the current direction to the pending direction"""
+        self.current_direction = self.next_direction
+
+    def _update_position(self):
+        """Update the stored position of the player in grid coordinates"""
+        grid_position = self.game.level.convert_to_grid_position(self.center)
+        self.grid_position = grid_position
+
+
+class PlayerBeetle(Character):
+
+    """Store methods and properties relevant to the player character.
+
+    This class stores methods and properties relating to only the player,
+    namely checking if the player has collided with an enemy or is dead.
+
+    Kivy Events:
+    on_grid_position -- check for enemy collisions
+    on_dead -- remove life if player is dead
+
+    Kivy Properties:
+    color -- ObjectProperty storing the color of the character
+    dead -- BooleanProperty storing whether the player is dead or not
+    """
+
+    start_x = NumericProperty(0)
+    start_y = NumericProperty(0)
+    start_position = ReferenceListProperty(start_x, start_y)
+
+    color = ObjectProperty((1, 1, 0))
+    dead = BooleanProperty(False)
+    powered_up = BooleanProperty(False)
+
+    last_chomp_high = BooleanProperty(False)
+
+    def on_grid_position(self, instance, value):
+        """Check for enemy and pellet collisions on grid position change.
+
+        When the player grid position changes, this kivy event is called
+        and checks if player has collided with an enemy or pellet.
+        """
+        self.__check_enemy_collision()
+        self.__check_pellet_collision()
+
+    def __check_pellet_collision(self):
+        """Check for pellet collision.
+
+        This method checks if the player has collided with any pellets.
+        If so, the score is adjusted or a power-up is applied appropriately.
+        """
+        current_cell = self.game.level.get_cell(self.grid_position)
+        if current_cell.pellet_exists:
+            current_cell.remove_pellet()
+            self.game.score += self.game.pellet_value
+
+            if not self.last_chomp_high:
+                self.game.sounds['chomp_high'].play()
```

Comment:
I notice significant slowdown whenever a sound is played. 
Wasn't sure if its a code issue or an issue with Kivy.